src/DbalStore.php

```diff
@@ -0,0 +1,305 @@
+            $results[$notResolved] = $default;
+        }
+
+        return $results;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getMultipleOrFail(array $keys)
+    {
+        KeyUtil::validateMultiple($keys);
+
+        // Normalize indices of the array
+        $keys = array_values($keys);
+
+        $data = $this->doGetMultiple($keys);
+
+        $results = array();
+        $resolved = array();
+        foreach ($data as $row) {
+            $results[$row['meta_key']] = Serializer::unserialize($row['meta_value']);
+            $resolved[] = $row['meta_key'];
+        }
+
+        $notResolvedArr = array_diff($keys, $resolved);
+
+        if (!empty($notResolvedArr)) {
+            throw NoSuchKeyException::forKeys($notResolvedArr);
+        }
+
+        return $results;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function remove($key)
+    {
+        KeyUtil::validate($key);
+
+        try {
+            $result = $this->connection->delete($this->tableName, array('meta_key' => $key));
+        } catch (Exception $e) {
+            throw WriteException::forException($e);
+        }
+
+        return $result === 1;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function exists($key)
+    {
+        KeyUtil::validate($key);
+
+        try {
+            $result = $this->connection->fetchAssoc('SELECT * FROM '.$this->tableName.' WHERE meta_key = ?', array($key));
+        } catch (Exception $e) {
+            throw ReadException::forException($e);
+        }
+
+        return $result ? true : false;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function clear()
+    {
+        try {
+            $stmt = $this->connection->query('DELETE FROM '.$this->tableName);
+            $stmt->execute();
+        } catch (Exception $e) {
+            throw WriteException::forException($e);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function keys()
+    {
+        try {
+            $stmt = $this->connection->query('SELECT meta_key FROM '.$this->tableName);
+            $result = $stmt->fetchAll(PDO::FETCH_COLUMN);
+        } catch (Exception $e) {
+            throw ReadException::forException($e);
+        }
+
+        return $result;
+    }
+
+    private function doInsert($key, $value)
+    {
+        $serialized = Serializer::serialize($value);
+
+        try {
+            $this->connection->insert($this->tableName, array(
+                'meta_key' => $key,
+                'meta_value' => $serialized,
+            ));
+        } catch (Exception $e) {
+            throw WriteException::forException($e);
+        }
+    }
+
+    /**
+     * The name for our DBAL database table.
+     *
+     * @return string
+     */
+    public function getTableName()
+    {
+        return $this->tableName;
+    }
+
+    /**
+     * Object Representation of the table used in this class.
+     *
+     * @return \Doctrine\DBAL\Schema\Table
```

Comment:
Yes, but you have to know your abc very well because it also has to be in the right order :mortar_board: 
https://www.youtube.com/watch?v=75p-N9YKqNo